torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -939,6 +940,31 @@ def test_new_subgroups_overlap_not_allowed(self):
                     ranks_per_subgroup_list=[[0], [1, 2], [1, 3]]
                 )
 
+        @require_backend({"nccl"})
+        @require_backends_available({"nccl"})
+        @skip_if_lt_x_gpu(2)
+        def test_average_parameters_nccl(self):
+            rank = dist.get_rank()
+            rank_to_GPU = self._init_multigpu_helper()
+            device_id = rank_to_GPU[rank][0]
+            group_nccl = dist.new_group(ranks=[0, 1], backend="nccl")
+
+            model = (
+                nn.Sequential(
+                    nn.Conv2d(3, 3, kernel_size=3, padding=1),
+                    nn.ReLU(),
+                    nn.Linear(1, 5, bias=False)
+                ).cuda(device_id)
+            )
+
+            for p in model.parameters():
+                p.data = torch.ones_like(p.data) * rank
+            model_averaging_utils.average_parameters(model=model, process_group=group_nccl)
+
+            # Every element should be the average of 0 and 1, i.e., 0.5.
+            for p in model.parameters():
+                self.assertEqual(p.data, torch.ones_like(p.data) * 0.5)
```

Comment:
What if this test is ran on a 4-GPU machine, then ranks 2 and 3 would have 2/3s as model param data, and they shouldn't participate in the averaging process because the group passed in is only for ranks 0 and 1? Then it doesn't seem like this assert would succeed.

Reply:
Yeah, there should have been a condition the current rank is in the subgroup.

Updated the test to test both global averaging and partial averaging.